lib/utils.js

```diff
@@ -1,252 +1,186 @@
+   * happens when running in the context of an application (as opposed to tests or migration).
    *
-   * [ { logIndex: 13,
-   *     transactionIndex: 0,
-   *     transactionHash: "0x999e51b4124371412924d73b60a0ae1008462eb367db45f8452b134e5a8d56c8",
-   *     blockHash: "0xe35f7c374475a6933a500f48d4dfe5dce5b3072ad316f64fbf830728c6fe6fc9",
-   *     blockNumber: 294,
-   *     address: "0xd6a2a42b97ba20ee8655a80a842c2a723d7d488d",
-   *     type: "mined",
-   *     event: "NewOrg",
-   *     args: { _avatar: "0xcc05f0cde8c3e4b6c41c9b963031829496107bbb" } } ]
+   * @param contractName
    */
-  if (!tx.logs || !tx.logs.length) {
-    throw new Error("getValueFromLogs: Transaction has no logs");
-  }
-
-  if (eventName !== undefined) {
-    for (let i = 0; i < tx.logs.length; i++) {
-      if (tx.logs[i].event === eventName) {
-        index = i;
-        break;
-      }
-    }
-    if (index === undefined) {
-      const msg = `getValueFromLogs: There is no event logged with eventName ${eventName}`;
-      throw new Error(msg);
-    }
-  } else if (index === undefined) {
-    index = tx.logs.length - 1;
-  }
-  if (tx.logs[index].type !== "mined") {
-    const msg = `getValueFromLogs: transaction has not been mined: ${tx.logs[index].event}`;
-    throw new Error(msg);
-  }
-  const result = tx.logs[index].args[arg];
+  static requireContract(contractName) {
+    try {
+      const myWeb3 = Utils.getWeb3();
 
-  if (!result) {
-    const msg = `getValueFromLogs: This log does not seem to have a field "${arg}": ${tx.logs[index].args}`;
-    throw new Error(msg);
-  }
-  return result;
-}
+      const artifact = require(`../migrated_contracts/${contractName}.json`);
+      const contract = new TruffleContract(artifact);
 
-/**
- * side-effect is to set web3.eth.defaultAccount.
- * throws an exception on failure.
- */
-export function getDefaultAccount() {
-  const web3 = getWeb3();
-  const defaultAccount = (web3.eth.defaultAccount =
-    web3.eth.defaultAccount || web3.eth.accounts[0]);
-
-  if (!defaultAccount) {
-    throw new Error("eth.accounts[0] is not set");
-  }
-
-  // TODO: this should be the default sender account that signs the transactions
-  return defaultAccount;
-}
-
-/**
- * Hash a string the same way solidity does, and to a format that will be properly translated into a bytes32 that solidity expects
- * @param str a string
- */
-export function SHA3(str) {
-  const result = `0x${abi.soliditySHA3(["string"], [str]).toString("hex")}`;
-  return result;
-}
-
-export const ExtendTruffleContract = superclass =>
-  class {
-    constructor(contract) {
-      // Make sure this contract is configured with the web3 provider and default config values
-      superclass.setProvider(getWeb3().currentProvider);
-      superclass.defaults({
-        from: getDefaultAccount(), // Use web3.eth.defaultAccount as the from account for all transactions
+      contract.setProvider(myWeb3.currentProvider);
```

Comment:
we could remove this code by instantiating an instance of ExtendTruffleContract here instead of TruffleContract

Reply:
I would prefer to make that a separate ticket to follow the one you recently merged.  Can we do that?

Reply:
sure